python-challenges/email-scheduler/devashmit-day19/scheduler.py

```
@@ -0,0 +1,46 @@
+def render(tpl,v):
+    for k,val in v.items():tpl=tpl.replace(f"{{{{{k}}}}}",val)
+    return tpl
+
+def load_recipients(path):
+    ext=Path(path).suffix.lower()
+    if ext==".csv":
+        with open(path,newline="",encoding="utf-8") as f:return list(csv.DictReader(f))
+    with open(path,encoding="utf-8") as f:return json.load(f)
+
+def send_email(to,subject,body,dry_run):
+    if dry_run:print(f"  [DRY RUN] → {to}: {subject[:50]}");logging.info(f"DRY RUN to={to}");return
```

Ruff (0.15.17)

[error] Multiple statements on one line (colon)

(E701)